code/modules/modular_computers/file_system/programs/budgetordering.dm

```diff
@@ -0,0 +1,279 @@
+		homing beacons or machinery housing any form of artificial intelligence."
+	var/blockade_warning = "Bluespace instability detected. Shuttle movement impossible."
+	/// radio used by the app to send messages on supply channel
+	var/obj/item/radio/headset/radio
+	/// var that tracks message cooldown
+	var/message_cooldown
+	var/list/loaded_coupons
+
+/datum/computer_file/program/budgetorders/proc/get_export_categories()
+	. = EXPORT_CARGO
+
+/datum/computer_file/program/budgetorders/proc/is_visible_pack(mob/user, paccess_to_check, var/list/access, var/contraband)
+	if(issilicon(user)) //Borgs can't buy things.
+		return FALSE
+	if(computer.obj_flags & EMAGGED)
+		return TRUE
+	else if(contraband) //Hide contrband when non-emagged.
+		return FALSE
+	if(!paccess_to_check) // No required_access, allow it.
+		return TRUE
+	if(isAdminGhostAI(user))
+		return TRUE
+
+	//Aquire access from the inserted ID card.
+	if(!length(access))
+		var/obj/item/card/id/D
+		var/obj/item/computer_hardware/card_slot/card_slot
+		if(computer)
+			card_slot = computer.all_components[MC_CARD]
+			D = card_slot?.GetID()
+		if(!D)
+			return FALSE
+		access = D.GetAccess()
+
+	if(paccess_to_check in access)
+		return TRUE
+
+	return FALSE
+
+/datum/computer_file/program/budgetorders/ui_data()
+	var/list/data = get_header_data()
+	data["location"] = SSshuttle.supply.getStatusText()
+	var/datum/bank_account/buyer = SSeconomy.get_dep_account(ACCOUNT_CAR)
+	var/obj/item/computer_hardware/card_slot/card_slot = computer.all_components[MC_CARD]
+	var/obj/item/card/id/id_card = card_slot?.GetID()
+	if(id_card?.registered_account)
+		if(ACCESS_HEADS in id_card.access)
+			requestonly = FALSE
+			buyer = SSeconomy.get_dep_account(id_card.registered_account.account_job.paycheck_department)
+	if(buyer)
+		data["points"] = buyer.account_balance
+	data["away"] = SSshuttle.supply.getDockedId() == "supply_away"
+	data["self_paid"] = self_paid
+	data["docked"] = SSshuttle.supply.mode == SHUTTLE_IDLE
+	data["loan"] = !!SSshuttle.shuttle_loan
+	data["loan_dispatched"] = SSshuttle.shuttle_loan && SSshuttle.shuttle_loan.dispatched
+	var/message = "Remember to stamp and send back the supply manifests."
+	if(SSshuttle.centcom_message)
+		message = SSshuttle.centcom_message
+	if(SSshuttle.supplyBlocked)
+		message = blockade_warning
+	data["message"] = message
+	data["cart"] = list()
+	for(var/datum/supply_order/SO in SSshuttle.shoppinglist)
+		data["cart"] += list(list(
+			"object" = SO.pack.name,
+			"cost" = SO.pack.cost,
+			"id" = SO.id,
+			"orderer" = SO.orderer,
+			"paid" = !isnull(SO.paying_account) //paid by requester
+		))
+
+	data["requests"] = list()
+	for(var/datum/supply_order/SO in SSshuttle.requestlist)
+		data["requests"] += list(list(
+			"object" = SO.pack.name,
+			"cost" = SO.pack.cost,
+			"orderer" = SO.orderer,
+			"reason" = SO.reason,
+			"id" = SO.id
+		))
+
+	return data
+
+/datum/computer_file/program/budgetorders/ui_static_data(mob/user)
```

Comment:
This was taken and used directly from the cargo console as you can imagine, but as I'm using the ID card's access to move over what cargo packs can and cannot be seen/purchased by an ID, this can be spoofed with an open UI. I'm thinking ideally I move this over to just UI_data instead but I can easily see that getting out of hand, love some feedback on how to handle that.